interactions/models/internal/cooldowns.py

```diff
@@ -387,6 +387,22 @@ async def get_cooldown_time(self, context: "BaseContext") -> float:
         cooldown = await self.get_cooldown(context)
         return cooldown.get_cooldown_time()
 
+    async def get_cooldown_time_with_key(self, key: Any, *, create: bool = False) -> float:
```

Comment:
I cant decide if these two added methods should be sync or async. 

They dont *need* to be async, but i can see why you chose to make them that way

Reply:
maybe they should be sync, people with dashboards might be able to hack it and give owners a button to reset it, using flask they can only use sync. However, that does make it harder to remember (reset with context is async, but reset with key isn't), but if they're using a correct IDE (like VSC) it will complain. I'm actually more in favor of the former but I won't make a commit until you decide.

Reply:
Make then sync

Reply:
ok

Reply:
Keep in mind that slapping `async` on a method with an already defined type return won't mean the type being returned is the same.
![image](https://user-images.githubusercontent.com/41456914/232284257-1b077aca-f7ad-406a-9921-7fd1e5a24fa5.png)

Reply:
yes, but it will complain if a sync function (the chances are more likely that one will write async than forget it) is awaited.